src/cascadia/ShellExtension/OpenTerminalHere.cpp

```diff
@@ -0,0 +1,150 @@
+//   Package or the release package. We'll need to return different text, icons,
+//   and use different commandlines depending on which one the user requested.
+// - Uses a C++11 "magic static" to make sure this is only computed once.
+// - If we can't determine if it's the dev build or not, we'll default to true
+// Arguments:
+// - <none>
+// Return Value:
+// - true if we believe this extension is being run in the dev build package.
+bool IsDevBuild()
+{
+    // use C++11 magic statics to make sure we only do this once.
+    static bool isDevBuild = []() -> bool {
+        try
+        {
+            const auto package{ winrt::Windows::ApplicationModel::Package::Current() };
+            const auto id = package.Id();
+            const std::wstring name{ id.FullName() };
+            // Does our PFN start with WindowsTerminalDev?
+            return name.rfind(L"WindowsTerminalDev", 0) == 0;
+        }
+        CATCH_LOG();
+        return true;
+    }();
+
+    return isDevBuild;
+}
+
+// Method Description:
+// - This method is called when the user activates the context menu item. We'll
+//   launch the Terminal using the current working directory.
+// Arguments:
+// - psiItemArray: a IShellItemArray which contains the item that's selected.
+// Return Value:
+// - S_OK if we successfully attempted to launch the Terminal, otherwise a
+//   failure from an earlier HRESULT.
+HRESULT OpenTerminalHere::Invoke(IShellItemArray* psiItemArray,
+                                 IBindCtx* /*pBindContext*/)
+{
+    DWORD count;
+    psiItemArray->GetCount(&count);
+
+    winrt::com_ptr<IShellItem> psi;
+    RETURN_IF_FAILED(psiItemArray->GetItemAt(0, psi.put()));
+
+    PWSTR pszName;
+    RETURN_IF_FAILED(psi->GetDisplayName(SIGDN_FILESYSPATH, &pszName));
+
+    {
+        wil::unique_process_information _piClient;
+        STARTUPINFOEX siEx{ 0 };
+        siEx.StartupInfo.cb = sizeof(STARTUPINFOEX);
+
+        const bool isDevBuild = IsDevBuild();
+        std::wstring cmdline = fmt::format(L"{}.exe -d \"{}\"", isDevBuild ? L"wtd" : L"wt", pszName);
+        RETURN_IF_WIN32_BOOL_FALSE(CreateProcessW(
+            nullptr,
+            cmdline.data(),
+            nullptr, // lpProcessAttributes
+            nullptr, // lpThreadAttributes
+            false, // bInheritHandles
+            EXTENDED_STARTUPINFO_PRESENT | CREATE_UNICODE_ENVIRONMENT, // dwCreationFlags
+            nullptr, // lpEnvironment
+            nullptr,
+            &siEx.StartupInfo, // lpStartupInfo
+            &_piClient // lpProcessInformation
+            ));
+    }
+
+    CoTaskMemFree(pszName);
+
+    return S_OK;
+}
+
+HRESULT OpenTerminalHere::GetToolTip(IShellItemArray* /*psiItemArray*/,
+                                     LPWSTR* ppszInfotip)
+{
+    // tooltip provided here, in this case none is provided
+    *ppszInfotip = NULL;
+    return E_NOTIMPL;
+}
+
+HRESULT OpenTerminalHere::GetTitle(IShellItemArray* /*psiItemArray*/,

```

Comment:
So, i had an idea.

It looks like every verb has its own _id_.

Can we just use different verbs in the dev package and the prod package to change what we display? Further, can we use those different verbs to _choose which thing to launch_? The shell extension would natively support wt _and_ wtd, and make the decision based on how it was invoked rather than where it lives

Reply:
Okay that's a neat idea